contracts/SDUtilityPool.sol

```diff
@@ -0,0 +1,906 @@
+     * @return totalEth The total ETH bonded by the operator
+     */
+    function getOperatorTotalEth(address operator) public view returns (uint256) {
+        (, , uint256 totalValidators) = ISDCollateral(staderConfig.getSDCollateral()).getOperatorInfo(operator);
+
+        // Real bonded ETH is 4, but we use 2 to be conservative
+        uint256 totalEth = totalValidators * 2 ether;
+        return totalEth;
+    }
+
+    function getOperatorLiquidation(address account) external view override returns (OperatorLiquidation memory) {
+        if (liquidationIndexByOperator[account] == 0) return OperatorLiquidation(0, 0, 0, false, false, address(0));
+        return liquidations[liquidationIndexByOperator[account] - 1];
+    }
+
+    function getLiquidationThreshold() external view returns (uint256) {
+        return (riskConfig.liquidationThreshold);
+    }
+
+    /**
+     * @dev Assumes fee has already been accrued up to the current block
+     * @param sdAmount The amount of the SD token to delegate
+     */
+    function _delegate(uint256 sdAmount) internal {
+        /* Verify `accrualBlockNumber` block number equals current block number */
+        if (accrualBlockNumber != block.number) {
+            revert AccrualBlockNumberNotLatest();
+        }
+
+        uint256 exchangeRate = _exchangeRateStoredInternal();
+
+        if (!IERC20(staderConfig.getStaderToken()).transferFrom(msg.sender, address(this), sdAmount)) {
+            revert SDTransferFailed();
+        }
+        uint256 cTokenShares = (sdAmount * DECIMAL) / exchangeRate;
+        delegatorCTokenBalance[msg.sender] += cTokenShares;
+        cTokenTotalSupply += cTokenShares;
+
+        emit Delegated(msg.sender, sdAmount, cTokenShares);
+    }
+
+    function _requestWithdraw(uint256 _sdAmountToWithdraw, uint256 cTokenToBurn) internal returns (uint256) {
+        if (requestIdsByDelegatorAddress[msg.sender].length + 1 > maxNonRedeemedDelegatorRequestCount) {
+            revert MaxLimitOnWithdrawRequestCountReached();
+        }
+        sdRequestedForWithdraw += _sdAmountToWithdraw;
+        delegatorWithdrawRequests[nextRequestId] = DelegatorWithdrawInfo(
+            msg.sender,
+            cTokenToBurn,
+            _sdAmountToWithdraw,
+            0,
+            block.number
+        );
+        requestIdsByDelegatorAddress[msg.sender].push(nextRequestId);
+        emit WithdrawRequestReceived(msg.sender, nextRequestId, _sdAmountToWithdraw);
+        nextRequestId++;
+        return nextRequestId - 1;
+    }
+
+    function _utilize(address utilizer, uint256 utilizeAmount) internal {
+        /* Verify `accrualBlockNumber` block number equals current block number */
+        if (accrualBlockNumber != block.number) {
+            revert AccrualBlockNumberNotLatest();
+        }
+        if (getPoolAvailableSDBalance() - sdRequestedForWithdraw < utilizeAmount) {
```

Comment:
This is okay to be done this way without considering accumulatedProtocolFee.

The difference is this is an atomic operation whereas //FinalizationOfWithdrawalRequests is not.

Would comment here that _utilize takes precedence over withdrawProtocolFee

Reply:
Check DM about DAG.